### PR TITLE
Fix etmdummy.h file not found during Linux/arm crosscomponent build

### PR DESCRIPTION
**Pull request by echesakovMSFT**
PR #15611 introduces `./build.sh arm release cross crosscomponent` failure

```
/home/dotnet-bot/coreclr/src/inc/eventtracebase.h:333:10: fatal error: 'clretwallmain.h' file not found
#include "clretwallmain.h"


In file included from /home/dotnet-bot/coreclr/src/utilcode/loaderheap.cpp:11:
/home/dotnet-bot/coreclr/src/inc/eventtracebase.h:207:10: fatal error: 'etmdummy.h' file not found
#include "etmdummy.h"
``` 

This PR changes generating Event Logging Infrastructure step for Linux/cross-architecture components and follows the steps for Linux/native components.
In other words, instead of simply copying `$__GeneratedIntermediate` directory (which is not enough since there are missing header files in src/inc) `build.sh` runs `genEventing` `genEventPipe` `genLttngProvider` Python scripts with `$__CrossCompIntermediatesDir/*` set as the targets.

@RussKeldorph @BruceForstall Can you please take a look?

### CODE REVIEW

**Comment by brianrob** [comment]
@nategraf can you please review this?

**Comment by echesakovMSFT** [comment]
Just wanted to mention that fixing this issue is needed to unblock dotnet/core-setup#3314

**Comment by nategraf** [comment]
Sorry for the delay. I was OOF Friday.

Can we avoid repeating code? For example, could the repeated code be placed in a function instead and only the assignment of variables be done outside that function?

**Comment by echesakovMSFT** [comment]
Did refactoring of build.sh as suggested @nategraf PTAL

File: build.sh [hunk 1]
```diff
@@ -189,53 +189,60 @@ restore_optdata()
...
+    echo "Laying out dynamically generated Event test files, etmdummy stub functions, and external linkages"
 
     __PythonWarningFlags="-Wall"
     if [[ $__IgnoreWarnings == 0 ]]; then
         __PythonWarningFlags="$__PythonWarningFlags -Werror"
     fi
 
+    $PYTHON -B $__PythonWarningFlags "$__ProjectRoot/src/scripts/genEventing.py" --inc $__OutputIncDir --dummy $__OutputIncDir/etmdummy.h --man "$__ProjectRoot/src/vm/ClrEtwAll.man" --testdir "$__OutputEventProviderDir/tests"
+    if  [[ $? != 0 ]]; then
+	exit
```
Comment (nategraf) [inline build.sh]: Nit: Indent level

File: build.sh [hunk 2]
```diff
@@ -189,53 +189,60 @@ restore_optdata()
...
-                echo "Laying out dummy event logging provider"
-                $PYTHON -B $__PythonWarningFlags "$__ProjectRoot/src/scripts/genDummyProvider.py" --man "$__ProjectRoot/src/vm/ClrEtwAll.man" --intermediate "$__GeneratedIntermediateEventProvider"
-                if  [[ $? != 0 ]]; then
-                    exit
-                fi
-                ;;
-        esac
+    # Event Logging Infrastructure
+    if [[ $__SkipCoreCLR == 0 || $__ConfigureOnly == 1 ]]; then
+	generate_event_logging_sources "$__IntermediatesDir" "the native build system"
```
Comment (nategraf) [inline build.sh]: Nit: Line indent looks too far back

**Comment by echesakovMSFT** [comment]
@nategraf indents should be good now

**Comment by echesakovMSFT** [comment]
@dotnet-bot test Tizen armel Cross Checked Innerloop Build and Test please

**Comment by echesakovMSFT** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test please

**Comment by echesakovMSFT** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

File: build.sh [hunk 3]
```diff
@@ -189,53 +189,60 @@ restore_optdata()
...
-            *)
-                echo "Laying out dummy event logging provider"
-                $PYTHON -B $__PythonWarningFlags "$__ProjectRoot/src/scripts/genDummyProvider.py" --man "$__ProjectRoot/src/vm/ClrEtwAll.man" --intermediate "$__GeneratedIntermediateEventProvider"
-                if  [[ $? != 0 ]]; then
-                    exit
-                fi
-                ;;
-        esac
+    # Event Logging Infrastructure
+    if [[ $__SkipCoreCLR == 0 || $__ConfigureOnly == 1 ]]; then
```
Comment (BruceForstall) [inline build.sh]: `__SkipCoreCLR` is either 0 or 1, and if 1,  we already returned (above), so this check against 0 should always be true, right?

File: build.sh [hunk 4]
```diff
@@ -189,53 +189,60 @@ restore_optdata()
...
+    echo "Laying out dynamically generated Event test files, etmdummy stub functions, and external linkages"
 
     __PythonWarningFlags="-Wall"
     if [[ $__IgnoreWarnings == 0 ]]; then
         __PythonWarningFlags="$__PythonWarningFlags -Werror"
     fi
 
+    $PYTHON -B $__PythonWarningFlags "$__ProjectRoot/src/scripts/genEventing.py" --inc $__OutputIncDir --dummy $__OutputIncDir/etmdummy.h --man "$__ProjectRoot/src/vm/ClrEtwAll.man" --testdir "$__OutputEventProviderDir/tests"
+    if [[ $? != 0 ]]; then
+        exit
```
Comment (BruceForstall) [inline build.sh]: Shouldn't all these `exit` be `exit 1` or `exit $?`?

File: build.sh (hunk 3 above)
Comment (echesakovMSFT) [reply build.sh]: @BruceForstall You are right, thank you for catching this!

@nategraf I am wondering what logic was meant to be implemented in the first place? In build.cmd there are only checks for `"%__BuildNative%"=="1" (` and `"%__DoCrossArchBuild%"=="1" (`. Should we do the same here?

File: build.sh (hunk 4 above)
Comment (echesakovMSFT) [reply build.sh]: @BruceForstall For consistency with other parts of build.sh, I do `exit 1`

File: build.sh [hunk 5]
```diff
@@ -189,53 +189,60 @@ restore_optdata()
...
+        Linux|FreeBSD)
+            echo "Laying out dynamically generated Event Logging Implementation of Lttng"
+            $PYTHON -B $__PythonWarningFlags "$__ProjectRoot/src/scripts/genLttngProvider.py" --man "$__ProjectRoot/src/vm/ClrEtwAll.man" --intermediate "$__OutputEventProviderDir"
+            if [[ $? != 0 ]]; then
+                exit
+            fi
+            ;;
+        *)
+            echo "Laying out dummy event logging provider"
+            $PYTHON -B $__PythonWarningFlags "$__ProjectRoot/src/scripts/genDummyProvider.py" --man "$__ProjectRoot/src/vm/ClrEtwAll.man" --intermediate "$__OutputEventProviderDir"
```
Comment (BruceForstall) [inline build.sh]: I guess you didn't introduce this, but it seems unfortunate to replicate 90% of this command line (and the three lines below). Maybe the `case` could just set a variable and do the "echo" and then the python command could follow.

Comment (echesakovMSFT) [reply build.sh]: @BruceForstall I guess we can defer refactoring till another PR. Fixing this issue is of higher priority

**Comment by sandreenko** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Innerloop Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked normal Build and Test
@dotnet-bot test Windows_NT x86_arm_altjit Checked Build and Test
@dotnet-bot test Windows_NT x64_arm64_altjit Checked Build and Test

**Comment by echesakovMSFT** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Innerloop Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked normal Build and Test

**Comment by echesakovMSFT** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Innerloop Build and Test

**Comment by sandreenko** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Innerloop Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked normal Build and Test

**Comment by echesakovMSFT** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Innerloop Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked normal Build and Test

**Comment by sandreenko** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Innerloop Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked normal Build and Test

**Pull request merged**